### ISSUE TITLE
sdcard installation

### ISSUE THREAD

**Issue by LiohMoeller** [openwrt-zipit/openwrt-pxa-target#6]
I am trying to install to a sdcard as described here: https://github.com/openwrt-zipit/openwrt-pxa-target/wiki/Installation
The same card worked, booting the flasher from a fat32 partition. Sadly it still always boot from the internal flash and there is not much storage left to install additional applications. I have even tried to apply the sdcard fix, without success.

Is there anything else I can try? Shouldn't the card be also visible in /proc/partitions or show up in dmesg while booting from the internal flash?

Is there anything I have to change in uboot to correct the boot order?

**Comment by deeice** [comment]
Yes, you should see something about the mmc in both dmesg and /proc/partitions.

cat /proc/partitions
major minor  #blocks  name
  ...
  31        4       2112 mtdblock4
 179        0    1927168 mmcblk0
 179        1    1925730 mmcblk0p1

I see this in dmesg after the coprocessor line.

[    2.893207] XScale iWMMXt coprocessor detected.
[    2.914723] mmc0: host does not support reading read-only switch, assuming write-enable
[    2.956209] input: pxabus:gpio-keys as /devices/platform/pxabus/pxabus:gpio-keys/input/input1
[    2.983266] mmc0: new SD card at address 1234
[    3.002963] pxa-rtc 40900000.rtc: setting system clock to 2000-01-01 00:00:20 UTC (946684820)
[    3.034932] mmcblk0: mmc0:1234 SA02G 1.84 GiB
[    3.061286]  mmcblk0: p1
[    3.119161] VFS: Mounted root (ext2 filesystem) readonly on device 179:1.
[    3.138208] Freeing unused kernel memory: 152K (c044c000 - c0472000)
[    4.491727] EXT2-fs (mmcblk0p1): warning: mounting unchecked fs, running e2fsck is recommended

Does dmesg show anything if you insert the SD after booting from the internal Flash?

I think you could stop it in uboot to check some things, but the zipit IRC logs are gone so it's hard to remember all the tricks.

**Comment by LiohMoeller** [comment]
![IMG20211229155739](https://user-images.githubusercontent.com/1267594/147675116-84f4c01d-68d9-4b11-afc6-8108d5beabb7.jpg)

This is what I see when I insert the card after booting from internal storage. It is a 64GB MicroSD card within a MiniSD Adapter.

**Comment by mozzwald** [comment]
I'm guessing the newer kernel supplied by openwrt doesn't work with the SD card you are using. This was a common issue; it works to flash the system but the newer kernel cannot access it. Your best option is to just find a card that works. Smaller cards tend to be more likely to work in my experience.

**Comment by deeice** [comment]
Hey Mozzwald!  Is the old IRC still live?  I got purged (prolly the owner finally noticed my "trump won" post or some other freedom of speech slip) so I had to abandon ship and slink over to libera.  We discussed this a bit over there, but otherwise not much happening.  You should pop in and pep it up.

deeice

________________________________
